pkg/kubectl/cmd/apply.go

```diff
@@ -683,7 +692,38 @@ type patcher struct {
 	openapiSchema openapi.Resources
 }
 
+// verifyDruRunSupport reads the OpenAPI to see if the given GVK
+// supports dryRun. Sending dryRun requests to apiserver that don't
+// support it will result in objects being unwillingly persisted.
+//
+// If the GVK can not be found (it can be a CRD or a resource coming
```

Comment:
I won't do this assumption specially for aggregated server. I would say it safer to say it does not support dry-run than assuming it does.

Reply:
Agreed. I guess my argument is that this piece of code is safer than what we have today (nothing is checked at all).